File: backend/Services/Parsers/StrandsParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class StrandsParser : BasicScoreResultParser
    {
        private ILogger<StrandsParser> _logger;

        public StrandsParser(ILogger<StrandsParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override bool Default => false;
        public override string GameName => "Strands";
        public override string? HelpText => null;
        // \uD83D\uDCA1 = 💡
        // \uD83D\uDD35 = 🔵
        // \uD83D\uDFE1 = 🟡
        protected override Regex Parser => new Regex($"{GameName} .+?(?<{ScoreGroup}>[\uD83D\uDCA1\uDD35\uDFE1\n]{{2,}})", RegexOptions.Singleline);
        protected override string? ExtraContent => null;
        public override string Url => "https://www.nytimes.com/games/strands";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            dailyResult.Score = parserResults.Groups[ScoreGroup].Value
                .Trim()
                .Replace("\n", string.Empty)
                .Split("💡")
                .Length - 1;

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/strands-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class StrandsParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Strands';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://www.nytimes.com/games/strands';
  
  // 💡 = \uD83D\uDCA1, 🔵 = \uD83D\uDD35, 🟡 = \uD83D\uDFE1
  protected readonly parser = new RegExp(
    `${this.gameName} .+?(?<${BasicScoreResultParser.SCORE_GROUP}>[\uD83D\uDCA1\uDD35\uDFE1\\n]{2,})`,
    's'
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (!hasGroup(match, BasicScoreResultParser.SCORE_GROUP)) {
      return dailyResult;
    }
    
    const content = getGroup(match, BasicScoreResultParser.SCORE_GROUP) || '';
    const cleaned = content.trim().replace(/\n/g, '');
    
    // Count hint usage (💡)
    dailyResult.score = cleaned.split('💡').length - 1;
    
    return dailyResult;
  }
}
